roles/ceph-dashboard/tasks/configure_dashboard.yml

```diff
@@ -211,11 +193,31 @@
       with_items: "{{ groups[iscsi_gw_group_name] }}"
       run_once: true
 
-- name: inject grafana dashboard layouts
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard grafana dashboards update"
-  delegate_to: "{{ groups[mon_group_name][0] }}"
-  run_once: true
-  changed_when: false
+- name: set dashboard endpoints and inject layouts
+  block:
+  - name: set grafana url
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-url {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ grafana_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
+
+  - name: set alertmanager host
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-alertmanager-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ alertmanager_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
+
+  - name: set prometheus host
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-prometheus-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ prometheus_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
```

Comment:
Why moving this into the a `containerized_deployment` block ?
We need this for non containerized deployment too.

Reply:
you're right, we need to set these info for both scenarios. I'm going to update the PR, thanks @dsavineau